File: arrays/283. Move Zeroes/moveZeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let index = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] !== 0) {
            temp = nums[index];
            nums[index] = nums[i];
            nums[i] = temp;
            index++;
        }
    }
};

File: arrays/283. Move Zeroes/moveZeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # Track where to put non zero elements
        index = 0
        
        # This sorts the array putting non zero numbers at the beginning
        # leaving zeros at the end
        for i in range(len(nums)):
            if nums[i] != 0:
                temp = nums[index]
                nums[index] = nums[i]
                nums[i] = temp
                index += 1